### DISCUSSION TITLE
Issues Exporting Filled/Smoothed EBSD data

### DISCUSSION THREAD

**Discussion by elegantEmily** [mtex-toolbox/mtex#826] [Ask Anything] [answered]
Currently I am using the command "ebsd_Hraw = fill(ebsdG, grains);" to fill unindexed points. It appears to change some of the mis-indexed points. Is there any way to tell the function only to change points below a certain IQ or CI value? I want to report the number of pixels I have changed and the cutoff value when I report the IPF and ROD maps.

I would also like to export this filled data, however the current "export_ang" function doesn't seem to like the EBSDhex variable the fill function creates. Is there a way to create an EBSD variable from the EBSDhex variable? Or another way to get the "export_ang" function to take the filled ebsd data.

Thanks in advance!
Emily B.

**Comment by kilir** [comment, marked as answer]
Hi
a) fill will only fill non-existent points, so if you delete some points which - based on your choice - should be gone, it should fill those:
```
somecondition = ebsd.prop > 9000;
ebsd(somecondition)=[]
%sum(somecondition) %number of deleted points
```
b) `ebsd(:)` should get the EBSDhex back to EBSD.

Cheers,
Rüdiger

**Reply by elegantEmily** [reply]
Currently I am not doing any other smoothing algorithm. however it does look like the fill function is changing some of the misindexed point. I have attached IPF maps of before and after filling. 

This is the script between the first IPF plot and the filled IPF plot.
ebsd_Hraw1 = ebsd_Hraw;
ebsd_Hraw = fill(ebsdG, grains);
mis_ang=5*degree;
grod = ebsd_Hraw.calcGROD(grains);
SS=specimenSymmetry('orthorhombic');

% x = TD(RD)  y = TLTD z = STD
oM.inversePoleFigureDirection=xvector;
colorx=oM.orientation2color(ebsd_Hraw('Generic BCC').orientations);
oM.inversePoleFigureDirection=yvector;
colory=oM.orientation2color(ebsd_Hraw('Generic BCC').orientations);
oM.inversePoleFigureDirection=zvector;
colorz=oM.orientation2color(ebsd_Hraw('Generic BCC').orientations);

Do you know why this misindexed pixels are being changed? 

The ebsd conversion worked great!

Thanks!
![EBSD_AfterFill](https://user-images.githubusercontent.com/60235316/110035550-71a4d800-7d01-11eb-8e37-94ad290a0246.jpg)
![EBSD_BeforeFill](https://user-images.githubusercontent.com/60235316/110035552-723d6e80-7d01-11eb-8855-4b9f43d440e4.jpg)

**Reply by kilir** [reply]
Hi,
what version of mtex are you using? Fill in the current version does not smooth orientations
Here's an example using the current one:
```

mtexdata forsterite
ebsd = ebsd(inpolygon(ebsd,[5100 5300 2000 2000]))
%
[grains,ebsd.grainId, ebsd.mis2mean]=ebsd.calcGrains;
% get rid of smalle non-indexed area so we have something to fill
ebsd(grains(grains.grainSize<=8),'notIndexed')=[]
[grains,ebsd.grainId ebsd.mis2mean]=ebsd.calcGrains;
plot(ebsd,ebsd.mis2mean.angle/degree); hold on
plot(grains.boundary); hold off
CLim(gcm,[0,3])

nextAxis
% fill points
ebsdf = fill(ebsd,grains)
grod = calcGROD(ebsdf,grains);
[grains,ebsdf.grainId ebsdf.mis2mean]=ebsdf.calcGrains;
plot(ebsdf,ebsdf.mis2mean.angle/degree); hold on
plot(grains.boundary); hold off
CLim(gcm,[0,3])

```

![something](https://user-images.githubusercontent.com/11697557/110103081-cc900b00-7da5-11eb-84df-552116178c0e.png)

Cheers,
Rüdiger

**Reply by elegantEmily** [reply]
I am using MTEX 5.2.8. Should I download MTEX 5.6.0 or 5.5.2?

**Reply by kilir** [reply]
Try the newest one.
Cheers,
Rdiger
